tests/unit_tests/ode_solver_unit_test/explicit_ode_solver.cpp

```diff
@@ -0,0 +1,220 @@
+    PHiLiP::Parameters::AllParameters all_parameters;
+    all_parameters.parse_parameters (parameter_handler); // copies stuff from parameter_handler into all_parameters
+
+    //ode solver parameters
+    all_parameters.ode_solver_param.ode_solver_type = PHiLiP::Parameters::ODESolverParam::ODESolverEnum::explicit_solver;
+    all_parameters.ode_solver_param.nonlinear_max_iterations = 500;
+    all_parameters.ode_solver_param.print_iteration_modulo = 100;
+    
+    const int rk_order = 3;
+    double expected_order = -double(rk_order);
+    double order_tolerance = 0.1; 
+    all_parameters.ode_solver_param.runge_kutta_order = rk_order;
+
+    const double adv_speed_x = 1.0, adv_speed_y = 0.0;
+    all_parameters.manufactured_convergence_study_param.manufactured_solution_param.advection_vector[0] = adv_speed_x; //x-velocity
+    all_parameters.manufactured_convergence_study_param.manufactured_solution_param.advection_vector[1] = adv_speed_y; //y-velocity
+
+    int n_time_calculations = 4;
+    const double dt_init = 2.5E-3;
+    double dt = dt_init;
+    const double refine_ratio = 0.5;
+    const double finalTime = 1.0;
+
+    dealii::ConvergenceTable convergence_table;
+
+    double L2_error_old = 0;
+    dealii::Vector<double> L2_error_conv_rate(n_time_calculations);
+
+    for (int refinement = 0; refinement < n_time_calculations; ++refinement){
+
+        //initial_time_step is not modified by explicit ODE solver
+        all_parameters.ode_solver_param.initial_time_step = dt;
+        std::cout << "Using time step = " << dt << std::endl;
+        std::cout << "refinement = " << refinement << std::endl;
+
+        all_parameters.ode_solver_param.output_solution_every_x_steps = int(finalTime/dt/10.0); //output 10 vtk files (if dt reaches finalTime exactly)
+
+        const unsigned int space_poly_degree = rk_order + 2;
+        std::shared_ptr < PHiLiP::DGBase<dim, double> > dg = PHiLiP::DGFactory<dim,double>::create_discontinuous_galerkin(&all_parameters, space_poly_degree, grid);
+        dg->allocate_system ();
+
+        //initial conditions
+        std::cout << "Implement initial conditions" << std::endl;
+        dealii::FunctionParser<dim> initial_condition;
+        std::string variables;
+        std::map<std::string,double> constants;
+        constants["pi"] = dealii::numbers::PI;
+        std::string expression_initial;
+        if (dim == 2){
+            variables = "x,y";
+            expression_initial = "exp( -( 20*(x-1)*(x-1) + 20*(y-1)*(y-1) ) )";
+        }
+        else if (dim == 1){
+            variables = "x";
+            //expression_initial = "exp(- 20 * (x-1) * (x-1))";	
+            expression_initial = "sin(2*pi*x/2.0)";	
+        }
+        initial_condition.initialize(variables,
+                expression_initial,
+                constants);
+        dealii::VectorTools::interpolate(dg->dof_handler,initial_condition,dg->solution);
```

Comment:
your initial condition can be added inside `src/physics/initial_conditions/initial_condition.cpp`, also take a look at somehow using `src/testing/flow_solver.cpp` to take care of all these standard tasks (create ODE, create DG, create grid, initialize solution etc) including advancing the solution in time, @donovan97 has used the flow solver within at least one of his newly implemented test cases for doing that